include/quic/connection.hpp

```diff
@@ -24,7 +24,7 @@ namespace oxen::quic
     class Client;
     class Handler;
 
-    class Connection : public std::enable_shared_from_this<Connection>
```

Comment:
@jagerman what's the motivation behind removing this? i was thinking it may be useful at some point to hold a weak pointer to the connection in `Stream.user_data`

Reply:
No reason we can't re-add it later if we want it, but for now it isn't in use.